src/torrent.cpp

```diff
@@ -10594,7 +10595,7 @@ bool is_downloading_state(int const st)
 		span<sha256_hash> v2_span(hashes);
 		m_ses.disk_thread().async_hash(m_storage, piece, v2_span, flags
 			, [self = shared_from_this(), hashes = std::move(hashes)]
-			(piece_index_t p, sha1_hash const& h, storage_error const& error)
+			(piece_index_t p, sha1_hash const& h, storage_error const& error) mutable
```

Comment:
if this lambda is not mutable, the captured `hashes` will be `const` and `std::move()` won't actually do anything. Making it mutable will allow to move it into the next call.